packages/terra-show-hide/src/_ShowHideButton.jsx

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable react/button-has-type */
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './_ShowHideButton.module.scss';
+
+const cx = classNames.bind(styles);
+
+const KEYCODES = {
+  ENTER: 13,
+  SPACE: 32,
+  TAB: 9,
+};
+
+const ButtonTypes = {
+  BUTTON: 'button',
+  SUBMIT: 'submit',
+  RESET: 'reset',
+};
+
+const propTypes = {
+  /**
+   * Whether or not the button should be disabled.
+   */
+  isDisabled: PropTypes.bool,
+  /**
+   * Callback function triggered when clicked.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Callback function triggered when button loses focus.
+   */
+  onBlur: PropTypes.func,
+  /**
+   * Callback function triggered when button gains focus.
+   */
+  onFocus: PropTypes.func,
+  /**
+   * Callback function triggered when key is pressed.
+   */
+  onKeyDown: PropTypes.func,
+  /**
+   * Callback function triggered when key is released.
+   */
+  onKeyUp: PropTypes.func,
+  /**
+   * Callback ref to pass into the dom element.
+   */
+  refCallback: PropTypes.func,
+  /**
+   * Sets the button text.
+   */
+  text: PropTypes.string.isRequired,
+  /**
+   * Sets the button type. One of `button`, `submit`, or `reset`.
+   */
+  type: PropTypes.oneOf([ButtonTypes.BUTTON, ButtonTypes.SUBMIT, ButtonTypes.RESET]),
+};
+
+const defaultProps = {
+  isDisabled: false,
+  refCallback: undefined,
+  type: ButtonTypes.BUTTON,
+};
+
+class Button extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { active: false, focused: false };
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.handleOnBlur = this.handleOnBlur.bind(this);
+  }
+
+  handleOnBlur(event) {
+    this.setState({ focused: false });
+
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  }
+
+  handleKeyDown(event) {
+    // Add active state to FF browsers
+    if (event.nativeEvent.keyCode === KEYCODES.SPACE) {
```

Comment:
Nitpick:

For maintainability, I feel it would be better to use an if else statement, and have the block for space do both the active and focused values on the components internal state. I think that's better than spreading the space across multiple conditionals.

Reply:
Agreed that is better logic: https://github.com/cerner/terra-core/pull/1956/commits/db61dc7961a60dfa0ef6406a735ae6c05d89d524